### PR TITLE
Base layout

### LINKED ISSUE #56: Apply base / layout styling
We need to get the layout / universal styling in the app at some point, might as well do that now.

This basically comes down to fonts, line spacing, and the overall layout (header / editor / sidebar) of the app.

### PR DESCRIPTION
**Pull request by slifty**
This adds some styling to what has been made so far to get it to match the designs, this is in preparation for adding functionality around the side pane.

image

Resolves #56

### CODE REVIEW

File: src/client/components/App.js [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
 import { Link, Route, BrowserRouter } from 'react-router-dom'
+import styled from 'styled-components'
 
 // App Imports
 import Header from './Header'
 import Sidebar from './Sidebar'
 import Editor from './Editor'
 
+// Styles
+const AppWrapper = styled.div`
```
Comment (reefdog) [inline src/client/components/App.js]: Is this just a hook for styling you'll apply later, or are you standardizing around generating major UI markup as components (whether styled or normal React)?

**Comment by reefdog** [comment]
Looks good. Over in TPT I followed the convention of prefixing `Styled` to styled components themselves, so rather than `AppWrapper` I'd have gone `StyledApp` (or maybe `StyledAppWrapper` for combined explicitness). I do like that as it distinguishes them from standard React components. It does get noisy though. What do you think?

File: src/client/components/App.js (hunk 1 above)
Comment (slifty) [reply src/client/components/App.js]: Yeah I saw that too -- I was screwing around at one point trying to get the application to fill 100% of the height, and this was part of the "smash my face into the problem" solutionin (there used to be styles in this element)

I forgot to delete it; I should probably go ahead and revert back to the <div> for now.

Comment (reefdog) [reply src/client/components/App.js]: Ah got it. Didja solve the 100% height or need a hand?

**Comment by slifty** [comment]
Ahh that is nice -- more explicit for sure.  I'll go add that in now.

**Comment by slifty** [comment]
Hmmm ok so the problem with the Styled prefix is that it might be TOO explicit in some cases.  For example:

```
        <EditorWrapper>
          <Backdrop
            ref="backdrop">
            <Highlights>
              {this.generateFlaggedText()}
            </Highlights>
          </Backdrop>
          <EditorTextArea/>
        </EditorWrapper>
```

I don't THINK I would want to prefix Styled to all of those things (even though they are all styled components).

I wonder if something even more generic would make sense, like "Root" or "StyledRoot" for cases when it is just a, well, root styled component.

**Comment by slifty** [comment]
Did some reading up on this.

[This article](https://medium.com/inturn-eng/naming-styled-components-d7097950a245) outlines an approach that uses an "S" namespace for styled components.  So for instance:

```
const S = {};
S.Navigation = styled.div`/* styles */`;
S.Content = styled.div`/* styles */`;
```

[This thread](https://github.com/styled-components/styled-components/issues/1604) basically says to just call the root wrapper "Wrapper" instead of "NavigationWrapper"

I'm not entirely sure which path I prefer so I'm going to... punt temporarily and just keep it as is, not use the Styled convention after all.  I like the idea of the namespace but I'm worried it will be one of those things where I do it and then two weeks later say "why did I get this triforce face tattoo"

**Pull request merged**